### DISCUSSION TITLE
Attached Properties as Property Match in Style Selector

### DISCUSSION THREAD

**Discussion by arnab741** [AvaloniaUI/Avalonia#15091] [Q&A] [answered]
I've an Attached Property on CheckBox and want to apply some Styles only when this property is enabled, based on the reference I get on https://docs.avaloniaui.net/docs/reference/styles/style-selector-syntax I tried creating the selector as follows but it doesn't work

**_<Style Selector="CheckBox[(vm:AttachedProps.Test)=0] /template/ Path" >_**

```c#
public class AttachedProps : AvaloniaObject
{
  public static readonly AttachedProperty<int> TestProperty =
    AvaloniaProperty.RegisterAttached<CheckBox, Control, int>("Test", defaultValue:0);

  public static int GetTest(Control element)
  {
    return element.GetValue(TestProperty);
  }

  public static void SetTest(Control element, int value)
  {
    element.SetValue(TestProperty, value);
  }
}
``` 

On a similar note, to workaround for my application I thought of conditionally applying style classes as MenuItems already have a DataTemplate defined, certainly can't overlap that, I was looking for some way to Add Style Classes here by Binding to ViewModel
Since, I believe Classes are Attached Properties 

Something like this or anything that can be used in this case for styling the menu item childrens
Example :: **_Setter Property="Classes.StyleC1" Value="{Binding Enable}"_**

```
<MenuItem Header="SampleItems"
                    Theme="{StaticResource StyleA}"
                    ItemsSource="{Binding ItemList}" >
  <MenuItem.ItemContainerTheme>
    <ControlTheme TargetType="{x:Type MenuItem}" BasedOn="{StaticResource Style2}" >
      <Setter Property="Header" Value="{Binding Name}" />
      <Setter Property="Command" Value="{Binding TestCommand}" />
      <Setter Property="CommandParameter" Value="{Binding $self}" />
    </ControlTheme>
  </MenuItem.ItemContainerTheme>
</MenuItem>
```

**Comment by jp2masa** [comment, marked as answer]
Did you try `<Style Selector="CheckBox[(vm|AttachedProps.Test)=0] /template/ Path">`?

**Reply by arnab741** [reply]
thanks ! that works, wasn't aware of that 

although for some reason i don't see the breakpoints hitting on the get/set accessors for the property, but works anyways
